packages/core/src/package/analyser/FHTAnalyzer.ts

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+import * as fs from 'fs-extra';
+import * as yaml from 'js-yaml';
+import { ComponentSet, registry } from '@salesforce/source-deploy-retrieve';
+import SfpPackage, { PackageType } from '../SfpPackage';
+import { PackageAnalyzer } from './PackageAnalyzer';
+
+export default class FHTAnalyser implements PackageAnalyzer{
+    
+    public async analyze(sfpPackage: SfpPackage): Promise<SfpPackage> {
+        let isFHTFieldFound = false;
+        let fhtFields: { [key: string]: Array<string> } = {};
+
+        //read the yaml
+        let fhtYamlPath = path.join(
+            sfpPackage.workingDirectory,
+            sfpPackage.projectDirectory,
+            sfpPackage.packageDirectory,
+            '/postDeploy/history-tracking.yaml'
+        );
+
+        //read components mentioned in yaml
+        if (fs.existsSync(fhtYamlPath)) {
+            //convert yaml to json
+            fhtFields = yaml.load(fs.readFileSync(fhtYamlPath, { encoding: 'utf-8' }));
+        }
+
+        //Load component Set
+        let componentSet = ComponentSet.fromSource(
+            path.join(sfpPackage.workingDirectory, sfpPackage.projectDirectory, sfpPackage.packageDirectory)
```

Comment:
check path exists

Reply:
not required as these are guaranteed from SfpPackage, it will be set to blank if not set

Reply:
test failing from this part 
<img width="549" alt="image" src="https://user-images.githubusercontent.com/29746243/208365342-fa618632-2ae5-46f9-98b2-1e545db95431.png">